src/ape/api/transactions.py

```diff
@@ -168,6 +188,14 @@ def ran_out_of_gas(self) -> bool:
             same amount of gas as the given ``gas_limit``.
         """
 
+    @property
+    def trace(self) -> Iterator[TraceFrame]:
+        """
+        The trace of the transaction, if available from your provider.
+        """
+
+        return self.provider.get_transaction_trace(txn_hash=self.txn_hash)
```

Comment:
We should find a way to cache this request, as it is really slow. Thoughts?

Reply:
[`itertools.tee`](https://docs.python.org/3/library/itertools.html#itertools.tee) is really nice as it let's you make copies of iterators for parallel usage. Perhaps suggest doing this for now (always keep a `tee` copy around in case we ask again), but eventually might need to get into the `QueryAPI` and caching it to cache db for long-term usage (with appropiate pruning).